### PR TITLE
Fix toggle state return from onToggle + add tests

### PR DESCRIPTION
**Pull request by CPatchane**
The toggle was always returning `!props.checked` instead of returning the real new toggle state.

### CODE REVIEW

File: react/Toggle/index.jsx [hunk 1]
```diff
@@ -4,9 +4,9 @@ import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 
 class Toggle extends Component {
-  onChange() {
+  onChange(e) {
     if (this.props.onToggle) {
-      this.props.onToggle(!this.props.checked)
+      this.props.onToggle(e.target.checked)
```
Comment (CPatchane) [inline react/Toggle/index.jsx]: Maybe it was used in another way by changing the props from the parent. I can't know for now.

File: react/Toggle/test/toggle.spec.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    expect(component).toMatchSnapshot()
+  })
+
+  it(`renders correctly if disabled`, () => {
+    const component = shallow(
+      <Toggle id="mock-toggle-id" disabled />
+    ).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
```
Comment (ptbrowne) [inline react/Toggle/test/toggle.spec.js]: les 3 cas au dessus sont gérés via les snap des exemples non ?

File: react/Toggle/index.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/Toggle/index.jsx]: The usage as a controlled component is shown in the styleguide.

File: react/Toggle/test/toggle.spec.js (hunk 2 above)
Comment (CPatchane) [reply react/Toggle/test/toggle.spec.js]: Indeed 👍

File: react/Toggle/index.jsx (hunk 1 above)
Comment (CPatchane) [reply react/Toggle/index.jsx]: Perfect so.

File: react/Toggle/test/toggle.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { configure, shallow } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-15'
+
+import Toggle from '../'
+
+configure({ adapter: new Adapter() })
+
+describe('Toggle component', () => {
+  beforeEach(() => {
+    jest.resetModules()
```
Comment (ptbrowne) [inline react/Toggle/test/toggle.spec.js]: why is it necessary ?

File: react/Toggle/test/toggle.spec.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+'use strict'
+/* eslint-env jest */
+
+import React from 'react'
+import { configure, shallow } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-15'
+
+import Toggle from '../'
+
+configure({ adapter: new Adapter() })
```
Comment (ptbrowne) [inline react/Toggle/test/toggle.spec.js]: the adapter is already configured in the jest.setup file

**Comment by ptbrowne** [comment]
Thanks @CPatchane 🌮 !

**Pull request merged**